source/tests/system/nisync_driver_api_tests.cpp

```diff
@@ -0,0 +1,388 @@
+    request.set_sync_clock(syncClock);
+    request.set_invert(invert);
+    request.set_update_edge(updateEdge);
+    request.mutable_vi()->set_id(driver_session_->id());
+    auto grpcStatus = GetStub()->ConnectTrigTerminals(&clientContext, request, &response);
+    *viStatusOut = response.status();
+    return grpcStatus;
+  }
+
+  ::grpc::Status call_DisconnectTrigTerminals(ViConstString srcTerminal, ViConstString destTerminal, ViStatus* viStatusOut)
+  {
+    ::grpc::ClientContext clientContext;
+    nisync::DisconnectTrigTerminalsRequest request;
+    nisync::DisconnectTrigTerminalsResponse response;
+    request.set_src_terminal(srcTerminal);
+    request.set_dest_terminal(destTerminal);
+    request.mutable_vi()->set_id(driver_session_->id());
+    auto grpcStatus = GetStub()->DisconnectTrigTerminals(&clientContext, request, &response);
+    *viStatusOut = response.status();
+    return grpcStatus;
+  }
+
+private:
+  std::shared_ptr<::grpc::Channel> channel_;
+  std::unique_ptr<::nidevice_grpc::Session> driver_session_;
+  std::unique_ptr<nisync::NiSync::Stub> nisync_stub_;
+  std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+  std::unique_ptr<nisync::NiSyncLibrary> nisync_library_;
+  std::unique_ptr<nisync::NiSyncService> nisync_service_;
+  std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(NiSyncDriverApiTest, ConnectClkTerminals_ReturnsSuccess)
+{
+  ViStatus viStatus;
+  auto srcTerminal = NISYNC_VAL_OSCILLATOR, destTerminal = NISYNC_VAL_CLKOUT;
+  auto grpcStatus = call_ConnectClkTerminals(srcTerminal, destTerminal, &viStatus);
+
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatus);
+  call_DisconnectSWTrigFromTerminal(srcTerminal, destTerminal, &viStatus);
+}
+
+TEST_F(NiSyncDriverApiTest, ConnectedClkTerminals_DisconnectClkTerminals_ReturnsSuccess)
+{
+  ViStatus viStatus;
+  auto srcTerminal = NISYNC_VAL_OSCILLATOR, destTerminal = NISYNC_VAL_CLKOUT;
+  auto grpcStatus = call_ConnectClkTerminals(srcTerminal, destTerminal, &viStatus);
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatus);
+
+  grpcStatus = call_DisconnectClkTerminals(srcTerminal, destTerminal, &viStatus);
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatus);
+}
+
+TEST_F(NiSyncDriverApiTest, ConnectedClkTerminals_DisconnectNotConnectedClkTerminals_ReturnsSuccess)
+{
+  ViStatus viStatus;
+  auto srcTerminal = NISYNC_VAL_OSCILLATOR, destTerminal = NISYNC_VAL_CLKOUT;
+  auto grpcStatus = call_ConnectClkTerminals(srcTerminal, destTerminal, &viStatus);
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatus);
+
+  auto notConnectedSrcTerminal = NISYNC_VAL_CLKIN;
+  grpcStatus = call_DisconnectClkTerminals(notConnectedSrcTerminal, destTerminal, &viStatus);
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatus);
+  call_DisconnectSWTrigFromTerminal(srcTerminal, destTerminal, &viStatus);
+}
+
+TEST_F(NiSyncDriverApiTest, ConnectSWTrigToTerminal_ReturnsSuccess)
```

Comment:
@tkrebes All the SWTrig tests are failing with error -1074118605 destTerminalInvalid. The values I'm using in the test are from the iotrace log where the call returned success. Any ideas what I'm doing wrong?
99.  niSync_ConnectSWTrigToTerminal (0x7FFFFFFF, "GlobalSoftwareTrigger", "PXIe_DStarC", "SyncClkFullSpeed", 1 (0x1), 0 (0x0), 0)
Process ID: 0x0000189C         Thread ID: 0x00004654
Start Time: 12:14:25.3614      Call Duration 00:00:00.0000
Status: 0 (VI_SUCCESS)

Reply:
I'd expect those values to work for an NI PXIe-6674T.

Reply:
I found that in some previous tests, I was incorrectly calling DisconnectSWTrigFromTerminal when I should be calling DisconnectClkTerminals and fixing those tests fixed this test as well. It's passing now.